### PR TITLE
【Hackathon 4th No.11】 为 paddle 新增 paddle.distribution.Geometric 分布

### PR DESCRIPTION
**Pull request by dasenCoding**
rfc : Geometric Distribution

### CODE REVIEW

**Comment by dasenCoding** [comment]
@luotao1  还有一个，多谢多谢！

File: rfcs/APIs/20230223_api_design_for_geometric.md [hunk 1]
```diff
@@ -0,0 +1,774 @@
...
+
+# 一、概述
+## 1、相关背景
+几何分布（`Geometric distribution`）是离散型概率分布。其中一种定义为：在n次伯努利试验中，试验k次才得到第一次成功的机率。详细地说，是：前k-1次皆失败，第k次成功的概率。几何分布是帕斯卡分布当r=1时的特例。[1]
+
+随着n增大呈等比级数变化，等比级数又称几何级数。这可能和以前几何学中无限分割图形得到的级数有关。
+
+`Geometric distribution` 应用条件：进行一系列独立试验，每一次试验活成功或失败，每一次试验的成功概率相同，即：为了取得第一次成功，需要进行多少次试验。
+
+目前 Paddle 框架中没有继承 Geometric 分布。所以此任务的目标是在 Paddle 框架中，基于现有概率分布方案，在其基础上进行扩展，新增 Geometric API，API 的调用路径为： `paddle.distribution.Geometric`。
```
Comment (cxxly) [inline rfcs/APIs/20230223_api_design_for_geometric.md:17]: 没有实现

File: rfcs/APIs/20230223_api_design_for_geometric.md [hunk 2]
```diff
@@ -0,0 +1,774 @@
...
+直接使用 Geometric 分布的名称作为此 API 的名称，参数保持 Geometric 分布最原生的参数即：
+
+- probs：伯努利实验成功概率
+- logits：采样的对数概率
+
+预期 paddle 调用 Geometric API 的形式为：
+```python
+#probs: the probability of sampling 1. Must be in range (0, 1]
+#logits：the log-odds of sampling 1.
+paddle.distribution.geometric.Geometric(probs, logits)
```
Comment (cxxly) [inline rfcs/APIs/20230223_api_design_for_geometric.md:625]: 这两个参数是二选一，还是必须都传

File: rfcs/APIs/20230223_api_design_for_geometric.md (hunk 1 above)
Comment (dasenCoding) [reply rfcs/APIs/20230223_api_design_for_geometric.md:17]: 收到！

File: rfcs/APIs/20230223_api_design_for_geometric.md (hunk 2 above)
Comment (dasenCoding) [reply rfcs/APIs/20230223_api_design_for_geometric.md:625]: > 这两个参数是二选一，还是必须都传；我理解是二选一，目前Paddle只需要支持probs就行；logits如果有兴趣可以所有分布统一支持

 1. 确实是二选一，是我没有表达清楚。
 2. logits 的实现确实是需要额外的处理。后面能力允许的话再尝试着如何让所有分布都支持 logits。

**Comment by dasenCoding** [comment]
@cxxly  您上面提出的修改意见都已进行修改，劳烦抽空再给review一下，感谢！

**Review by cxxly — APPROVED** [review]
LGTM

**Comment by jzhang533** [comment]
我们已经收到了你的申诉邮件，请等待回复。
请不要在PR comments中讨论与PR无关的事情。

**我的这个回复是一个警告，如果继续在PR comments里回复此类内容，我们会考虑对你在这里的发言做进一步的限制。**
